src/csharp/Grpc.HealthCheck.Tests/HealthServiceImplTest.cs

```diff
@@ -83,6 +84,43 @@ public void NullsRejected()
             Assert.Throws(typeof(ArgumentNullException), () => impl.ClearStatus(null));
         }
 
+#if GRPC_SUPPORT_WATCH
+        [Test]
+        public async Task Watch()
+        {
+            var cts = new CancellationTokenSource();
```

Comment:
it would be good to add a test that demonstrates what happens if the service status changes faster than the Watch method can send them out.

Reply:
Done